proposals/pubsub.md

```diff
@@ -0,0 +1,182 @@
+- Disabled pubsub makes an ecosystem more complex.
+- Building (near) real-time application on IPFS requires pubsub.
+
+#### User workflow example
+_How would a developer or user use this new capability?_
+<!--(short paragraph)-->
+
+They would just be able to use `pubsub` without `--enable-pubsub-experiment` flag. Feature becomes usable to ipfs-client's that use IPFS HTTP API.
+
+> **From Fission**:
+> 
+> If we expect apps to be independent and run on the IPFS network -- like they do with HTTP servers -- then whatever else can be made available natively can be powerful building blocks
+>
+> Having more capabilities available out of the box for developers is a better reason to build and host on IPFS natively. Pubsub out of the box gives a host of real time and messaging opportunities that our customers would love to use.
+
+#### Impact
+_How would this directly contribute to web3 dev stack product-market fit?_
+
+<!--
+Explain how this addresses known challenges or opportunities.
+What awesome potential impact/outcomes/results will we see if we nail this project?
+-->
+
+- PubSub becomes more reliable with more nodes on the network.
+- Bulding (near) real-time products becomes easier, IPFS nodes are suitable out of the box.
+
+#### Leverage
+_How much would nailing this project improve our knowledge and ability to execute future projects?_
+
+<!--
+Explain the opportunity or leverage point for our subsequent velocity/impact (e.g. by speeding up development, enabling more contributors, etc)
+-->
+
+- Improves velocity of teams building products requiring pubsub by removing additional burden.
+- Improves pubsub reliability by adding more enabled nodes to the network.
+- Enables development of new features that require pubsub.
+
+#### Confidence
+_How sure are we that this impact would be realized? Label from [this scale](https://medium.com/@nimay/inside-product-introduction-to-feature-priority-using-ice-impact-confidence-ease-and-gist-5180434e5b15)_.
+
+<!--Explain why this rating-->
+
+Level 3
+
+- We know teams building products that depend pubsub.
+- Features like IPNS over pubsub depends on this.
+
+
+## Project definition
+#### Brief plan of attack
+
+<!--Briefly describe the milestones/steps/work needed for this project-->
+
+- Measure what is the overhead on nodes that do not directly use pubsub
+- If overhead is neglegable ship a release of go-ipfs with pubsub enabled (see [ipfs/go-ipfs#6621][])
+- Enable / upgrade IPFS nodes operated by PL.
+- Raise awareness (blog, tweet, support community transition)
+
+#### What does done look like?
+_What specific deliverables should completed to consider this project done?_
+
+- Shipped release of go-ipfs with pubsub enabled
+- Increased number of pubsub enabled IPFS nodes in the network.
+
+####  What does success look like?
+_Success means impact. How will we know we did the right thing?_
+
+- Various projects (like OrbitDB) no longer have to explain with instructions how to enable pubsub.
```

Comment:
I believe @andrew's Forrest is also blocked by this

```suggestion
- Various projects (like OrbitDB, [Forest](https://github.com/forestpm/forest)) no longer have to explain with instructions how to enable pubsub.
```

Reply:
@andrew could you please confirm that ?

Reply:
yes partly, if someone is already running an ipfs node (via ipfs desktop for example) when they start forest, it will connect to that (otherwise starts it's own with pubsub), and if that existing node dosn't have pubsub enabled then some major parts of forest don't work.

Reply:
Thanks for clarifying. I'll just add Forest as per suggestion. While nuanced overall claim made is still valid.